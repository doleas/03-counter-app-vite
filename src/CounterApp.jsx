import {useState} from 'react'

import PropTypes from 'prop-types';



const CounterApp = ({value}) => {
    //Separamos la fucion del boton, esta definida de forma tradicional
    // function handleAdd(event){
    //     console.log(event)
    // }

    //Hooks 
    const [counter, setCounter] = useState(value);

    //funcion de flecha, se recomienda que si no utiliza nada del componente se la saque a fuera
    const handleAdd = (event) => {
        // return console.log(event)
        // setCounter(counter +1);
        setCounter((c)=>c+120);
    }

    return (  
        <>
            <h1> Counter App</h1>
            <h3> {value}</h3>
            {/* Esto es un contador */}
            {/* <button onClick={function(){console.log('+1')}}> Contador </button> */}
            {/* Pero con event puedo tener toda la informacion del evento como posicion... */}
            <button onClick={handleAdd}> Contador </button>
            {/* (event) => handleAdd(event) se lo puede reemplazar por solo manda la referencia a la funcion handleAdd */}

            <h3> {counter}</h3>

        </>
    )
}
 
export default CounterApp;

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 5
}