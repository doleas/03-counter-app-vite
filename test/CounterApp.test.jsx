import { render, screen } from "@testing-library/react";
import CounterApp from "../src/CounterApp";


describe('pruebas en <CounterApp />', () => {

    const initialValue = 10;

    test('debe hacer match con el el snappshot', () => { 
        
        const { container } = render(<CounterApp value={ initialValue }/>);

        expect(container).toMatchSnapshot();
    
    });

    test('debe de mostrar el valor inicial de 100', () => { 
        render(<CounterApp value={ 100 }/>);
        expect( screen.getAllByText(100) ).toBeTruthy();

        expect(screen.getByRole('heading', { level: 2}).innerHTML).toContain('100');
    });

});


