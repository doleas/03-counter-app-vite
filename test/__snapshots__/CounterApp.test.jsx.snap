// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <CounterApp /> debe hacer match con el el snappshot 1`] = `
<div>
  <h1>
     Counter App
  </h1>
  <h3>
     
    10
  </h3>
  <button>
     Contador 
  </button>
  <h2>
     
    10
  </h2>
</div>
`;
